crates/pallet-domains/src/lib.rs

```diff
@@ -455,6 +515,62 @@ mod pallet {
             Ok(())
         }
 
+        #[pallet::call_index(7)]
+        #[pallet::weight(Weight::from_all(10_000))]
+        // TODO: proper benchmark
+        pub fn submit_bundle_v2(
+            origin: OriginFor<T>,
+            opaque_bundle: OpaqueBundleOf<T>,
+        ) -> DispatchResult {
+            ensure_none(origin)?;
+
+            log::trace!(target: "runtime::domains", "Processing bundle: {opaque_bundle:?}");
+
+            let domain_id = opaque_bundle.domain_id();
+            let bundle_hash = opaque_bundle.hash();
+            let extrinsics_root = opaque_bundle.extrinsics_root();
+            let operator_id = opaque_bundle.operator_id();
+            let (bundle_author, receipt) = opaque_bundle.deconstruct();
+
+            let receipt_type = execution_receipt_type::<T>(domain_id, &receipt);
```

Comment:
I think this was called multiple times during the same function call IIUC

Reply:
This is the only one inside the `submit_bundle_v2` as we pass the `receipt_type` to `process_execution_receipt`, the other 2 calls are in `pre_dispatch` and `validate_unsigned`